### ISSUE TITLE
[CONCEPT] "Concept", "Scope", "Term" and "Inheritance of Terminology"

### ISSUE THREAD

**Issue by RieksJ** [trustoverip/concepts-and-terminology#4]
## Issue/Feature Description

When describing/discussing proposals and other issues, I find it necessary to be able to refer to concepts by name, rather than by (numerical) identifier. For example, I would like to be able to use a text such as "A single concept may be referred to by multiple Terms." rather than "A single Concept(00013) may be referred to by multiple Concept(00014)". 

## Proposed Solution

I've been thinking about this, and my suggestions go a bit beyond solving just the identified issue; perhaps this should be integrated in the C&T WG proposal?

**Requirements for scopes**:
* **we create a template for defining named scopes** (perhaps along the lines of 'glossary' in the WG proposal), one of which would be 'ToIP', and another one might be 'C&T'. 
* **a scope definition must identify its governing body**. For the 'C&T' scope, that would be the ToIP C&T WG.
* **a scope definition must state its purposes and objectives** that are being pursued (that might be a charter?) This allows to limit the work to whatever serves such purposes or helps realize such objectives, and reject everything else.
* **within a scope, Terms shall be defined and used for reasoning and communicating** about topics, issues, etc. within that scope. I think that for these purposes, we may not only need Terms that specifically refer to a (defined) Concept, but also Terms that refer to an existing definition somewhere 'in the wild'. 
* **within a scope, a (versioned) glossary of its terms shall be created and maintained**.
* **within a scope, other artefacts shall be (automatically) created and maintained**, for the purpose of assisting people as they reason and/or communicate within the scope's settings. Such artefacts may take various forms (web pages, pdfs, …), describe different things (e.g. use-cases, business models, research, conceptual models, etc.), etc. The decision of which artefacts are to be created and maintained lies with the scope-governors/conveners, which decide this given the scope's purposes/objectives and needs.

**Requirements for Concepts**:
* **concepts shall be defined within a scope** (and the same goes for the relations between them, constraints etc.). This implies that its criteria can be interpreted in the context of the purposes and objectives of that scope. Also, the relevance can be limited to that context.
* **concepts shall be associated with a term within the scope in which it is defined** (and the same goes for the relations between them, constraints etc.).  This allows for describing conceptual models that are relevant for that scope using the defined terms rather than numerical identifiers to refer to such concepts. This does not preclude other Terms to defined for such concepts.
* **concepts shall provide a description or proof of their relevance** that argue their need, and can also be used to determine whether or not concepts that are barely different should be treated differently, or can be made the same. 

**Requirements for Terms**
* **every term shall be defined within a scope**, which means that a reference to a concept definition (or another description/prescription that exists somewhere 'in the wild')  be associated with it, that provides its meaning within that scope. 

**Inheritance of Terminology**
* **(the governing body of) a scope may decide to inherit the terminology of other scopes**. When doing this, it must provide the rules for resolving conflicts (e.g. if two terminologies both define 'Identity', it must be clear which of these takes precedence). Obviously, terms defined in the scope itself have precedence over imported terms.

**Comment by vinomaster** [comment]
We need to separate Content Creation (Contribution, AutoGeneration) processing FROM Content approval lifecycle status such as working, preferred, accepted, superseded and deprecated. 

I would  suggest that the WG Charter explicitly call for the development of a clearly articulated content contribution lifecycle . This Process becomes a work product that is captured in a markdown file in  the  repo. so that contributors can understand process. 

Next Steps:

1. I will create a sample entry for the content contribution lifecycle process.
2. WG members can contribute the process content. 
3. I will make sure the Contributing Content Instructions refer to this process document.

**Comment by vinomaster** [comment]
The WG members now need to document the [content lifecycle process](https://github.com/trustoverip/concepts-and-terminology-wg/blob/master/Content_Lifecycle.md) and then close this issue.

**Comment by dhh1128** [comment]
I'm not understanding the difference between a "scope" and a "term". I'm finding myself thoroughly confused by the introduction of a "scope proposal" template that appears to exactly duplicate the "term proposal" template.

**Comment by dhh1128** [comment]
After re-reading this more carefully, I think I understand the proposed intent of a scope, but I disagree with the attempt to define them so strictly. I don't believe scopes are crisply definable; they are only approximately knowable because they are highly fluid and subjective. I prefer to define concepts carefully, and to *associate* them with a scope by tagging them for a knowledge domain and/or a community -- but not to try to fit them exactly into scope buckets in a super disciplined way.

**Comment by dhh1128** [comment]
BTW, I also disagree with "inheritance" as being too strict in its conception. Inheritance implies hierarchy. Hierarchy is controversial. (Is "sharia law" or "mosiac law" a subtype of a legal system, a subtype of a religious worldview, or both, or something messier?) What we want instead is a loose semantic graph with connections that can represent but do not require hierarchy. This will be much more able to tolerate fuzzy/messy/fluid situations.

**Comment by RieksJ** [comment]
w.r.t. "inheritance", the idea that I refer to has nothing to do with hierarchy. Rather it is the ability to import terms from multiple other sources; it's more like what in Object Oriented is called [multiple inheritance](https://en.wikipedia.org/wiki/Multiple_inheritance). Relabeling it to 'import' may perhaps solve this.

**Comment by RieksJ** [comment]
@dhh1128: I agree that scopes (in general) are much more fluid than the 'buckets' I suggest to define. Isn't that the idea of our terminology work, that we on the one hand allow people to fluidly/colloquially use words, and on the other hand we want to establish a precisely defined set of concepts that allow us to (also formally) reason?

I want to have a 'bucket scope' so that I can place related concepts that I carefully define in that bucket, and associate terms with them, for the purpose of (formally) reasoning with them within this single scope. Within a bucket scope, I can then create patterns (consisting of concepts, relations between the concepts, and constraints over them) that enable me to reason in a coherent fashion. In larger, perhaps more general and fluid scopes, I can compose patterns (perhaps from different bucket scopes) into a solid mental basis for realizing my objectives and addressing issues. 

More practically perhaps, it allows us to do away with syntax such as "Concept 2948 ("bird")" that is used in the concepts-issue-template. We can simply use (e.g.) {{bird}} to refer to Concept 2948 if 'bird' is a concept defined in the same scope as the concept whose definition is being proposed by the concept-issue. Or as {{somescope:bird}} in case 'bird' is defined in (bucket)scope 'somescope'.

Does that make sense to you?
